File: Data Structures & Algorithms/max-water-container/submission-5.cs
public class Solution {
    public int MaxArea(int[] heights) {
        int result = 0;
        int left = 0, right = heights.Length - 1;

        while (left < right)
        {
            int minVal = Math.Min(heights[left], heights[right]);
            int currentValue = minVal * (right - left);
            result = Math.Max(result, currentValue);

            if (heights[left] < heights[right])
                left++;
            else
                right--;
        }

        return result;
    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-5.ts
class Solution {
    /**
     * @param {string} s
     * @return {number}
     */
    lengthOfLongestSubstring(s: string): number {
        let i = 0
        let j = 0
        let result = 0
        const set = new Set()

        while (j < s.length) {
            if (set.has(s[j])) {
                set.delete(s[i])
                i++
            } else {
                result = Math.max(result, j - i + 1)
                set.add(s[j])
                j++
            }
        }

        return result
    }
}
